bench/clblas_benchmark.cpp

```diff
@@ -0,0 +1,359 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename clblas_benchmark.cpp
+ *
+ **************************************************************************/
+
+#include "blas_benchmark.hpp"
+
+#include <complex>
+#include <vector>
+
+#include <clBLAS.h>
+
+#include "clwrap.hpp"
+
+#define UNPACK_PARAM using ScalarT = TypeParam;
+
+#define CLBLAS_FUNCTION(postfix)                          \
+  template <typename T>                                   \
+  struct clblasX##postfix;                                \
+  template <>                                             \
+  struct clblasX##postfix<float> {                        \
+    static constexpr const auto func = &clblasS##postfix; \
+  };                                                      \
+  template <>                                             \
+  struct clblasX##postfix<double> {                       \
+    static constexpr const auto func = &clblasD##postfix; \
+  };
+/* template <> struct clblasX##postfix<std::complex<float>> { \ */
+/*   static constexpr const auto value = clblasC##postfix; \ */
+/* }; \ */
+/* template <> struct clblasX##postfix<std::complex<double>> { \ */
+/*   static constexpr const auto value = clblasZ##postfix; \ */
+/* }; */
+
+#define CLBLASI_FUNCTION(postfix)                          \
+  template <typename T>                                    \
+  struct clblasiX##postfix;                                \
+  template <>                                              \
+  struct clblasiX##postfix<float> {                        \
+    static constexpr const auto func = &clblasiS##postfix; \
+  };                                                       \
+  template <>                                              \
+  struct clblasiX##postfix<double> {                       \
+    static constexpr const auto func = &clblasiD##postfix; \
+  };
+/* template <> struct clblasiX##postfix<std::complex<float>> { \ */
+/*   static constexpr const auto value = clblasiC##postfix; \ */
+/* }; \ */
+/* template <> struct clblasiX##postfix<std::complex<double>> { \ */
+/*   static constexpr const auto value = clblasiZ##postfix; \ */
+/* }; */
```

Comment:
It declares a c++ interface to complex float and complex double functions in clblas, not very useful for benchmarking.